File: 프로그래머스/3/49191. 순위/순위.swift
import Foundation

func solution(_ n:Int, _ results:[[Int]]) -> Int {
    
    var adj:[[Int]] = [[Int]](repeating: [Int](repeating: 0, count: n+1), count: n+1)
    
    for result in results {
        
        let u = result[0]
        let v = result[1]
        
        adj[u][v] = 1
        
    }
    
    var count:Int = 0
    
    for k in 1...n {
        
        for i in 1...n {
            
            for j in 1...n {
                
                if (adj[i][k] == 1 &&  adj[k][j] == 1) {
                    
                    adj[i][j] = 1
                }
                
                
            }
            
        }
        
    }
    
    print(adj)
    
    for i in 1...n {
        
        var cnt:Int = 0
        
        for j in 1...n {
            if adj[i][j] == 1 || adj[j][i] == 1 {
                cnt += 1
            }
        }
        
        if cnt == n-1 {
            count += 1
        }
        

    }
    
    return count
}

File: 프로그래머스/3/49191. 순위/순위.kt
class Solution {
    fun solution(n: Int, results: Array<IntArray>): Int {
        var answer = 0
        val adj:Array<Array<Int>> =Array(n+1,{Array<Int>(n+1,{0})})
        
        for((u,v) in results)
        {
            adj[u][v]=1
        }
        
        for(k in 1..n)
        {
            for(u in 1..n)
            {
                for(v in 1..n)
                {
                    if(adj[u][k]==1&&adj[k][v]==1)
                    {
                        adj[u][v]=1
                    }
                }
            }
        }
        
        for(i in 1..n)
        {
            var cnt:Int =0
            for(j in 1..n)
            {
                if(adj[i][j]==1||adj[j][i]==1) cnt++
            }
            if(cnt==n-1) answer++
        }
        
        
        return answer
    }
}